diff --git a/meson.build b/meson.build
index 6dc157d..9aa2b84 100644
--- a/meson.build
+++ b/meson.build
@@ -229,6 +229,6 @@ meson.add_install_script('sh', '-c', 'mkdir -p ${DESTDIR}/' + sysconfdir + '/clo
 test(
   'py3',
   python,
-  args: ['-m', 'pytest', meson.project_source_root() / 'tests' / 'unittests'],
+  args: ['-m', 'pytest', '-k', 'not test_create_users', meson.project_source_root() / 'tests' / 'unittests'],
   timeout: 0,
-)
+)
\ No newline at end of file
